enterprise/jakarta.transformer/src/org/netbeans/modules/fish/payara/jakarta/transformer/TransformerWizardIterator.java

```diff
@@ -0,0 +1,505 @@
+                DialogDisplayer.getDefault().notify(notifyDescriptorTarget);
+            }
+            
+        }
+        
+        //override cases        
+        if(source != null && override) {
+            File sourceFile = new File(source);
+            
+            if(!sourceFile.exists()) {
+                DialogDisplayer.getDefault().notify(notifyDescriptorSource);
+            }
+            
+            //override folder on the same location
+            if(sourceFile.exists() && sourceFile.isDirectory()) {
+                return transformFolderOnSameLocation(resultSet, source);
+            } 
+            
+            //override file on the same location
+            if(sourceFile.exists() && sourceFile.isFile()) {
+                transformFileOnSameLocation(source);
+            }
+        }
+                               
+        return Collections.emptySet();
+    }
+    
+    public void transformFileOnSameLocation(String source) throws IOException {
+        File fileSource = new File(source);
+        File parentDirectory = new File(fileSource.getParent());
+        String pomFilePath = parentDirectory.getAbsolutePath()
+                + File.separator + TransformerConstants.POM_FILE_NAME;
+        File pomFile = new File(pomFilePath);
+        if (parentDirectory.exists() && parentDirectory.isDirectory()) {
+            FileObject directoryObject = FileUtil.toFileObject(parentDirectory);
+            if (!pomFile.exists()) {
+                copyTempPomFile(directoryObject);
+            }
+        }
+
+        FileObject p = FileUtil.toFileObject(parentDirectory);
+        if (p != null) {
+            Project project = ProjectManager.getDefault().findProject(p);            
+            requestProcessor.post(() -> {
+                NetbeansActionMapping mapping = new NetbeansActionMapping();
+                List<String> goals = new ArrayList<>();
+                goals.add(getLastTransformerVersionOfPlugin(TRANSFORMER_ARCHETYPE));
+                mapping.setGoals(goals);
+                mapping.setActionName("Transform source");
+                mapping.addProperty(TransformerConstants.SELECTED_SOURCE_PROPERTY_NAME,
+                        source);
+                mapping.addProperty(TransformerConstants.SELECTED_TARGET_PROPERTY_NAME,
+                        source);
+                ModelRunConfig rc = new ModelRunConfig(project, mapping,
+                        mapping.getActionName(), null,
+                        Lookup.EMPTY, false);
+                RunUtils.run(rc);
+            });
+
+            requestProcessor.post(() -> {
+                if (pomFile.exists()) {
+                    pomFile.delete();
+                }
+                File folderToDelete = new File(parentDirectory.getAbsolutePath()
+                        + File.separator + TransformerConstants.TARGET_FOLDER_NAME);
+                deleteFileTree(folderToDelete);
+            }, 8000);
```

Comment:
why is the second task separate? Can't the first task cleanup the files? Waiting a hardcoded amount of time before starting a cleanup is a bit of a red flag.

(applies to the other methods too)

The methods are very similar, would be good to make this more D.R.Y. if possible. Reuse code etc.

Reply:
> why is the second task separate? Can't the first task cleanup the files? Waiting a hardcoded amount of time before starting a cleanup is a bit of a red flag.
> 
> (applies to the other methods too)
> 
> The methods are very similar, would be good to make this more D.R.Y. if possible. Reuse code etc.

I will check this, to improve functionality